components/base/wheeled/wheeled_base.go

```diff
@@ -68,9 +68,9 @@ func (config *Config) Validate(path string) ([]string, error) {
 func init() {
 	wheeledBaseComp := registry.Component{
 		Constructor: func(
-			ctx context.Context, deps registry.Dependencies, config config.Component, logger golog.Logger,
+			ctx context.Context, deps registry.Dependencies, cfg config.Component, logger golog.Logger,
 		) (interface{}, error) {
-			return CreateWheeledBase(ctx, deps, config.ConvertedAttributes.(*Config), logger)
+			return createWheeledBase(ctx, deps, cfg, logger)
```

Comment:
You used to grab the ConvertedAttributes out of the config, cast it to a Config, and pass that to createWheeledBase. Now, you pass the whole config in, and the first line of CreateWheeledBase grabs the ConvertedAttributes out of it, casts to a Config, and ignores the rest. Why did this change? I'm not opposed to it, but I don't see the point and the old way looked simpler to me (and had simpler tests) because you were passing in only the data that would actually be used.

Reply:
our constructor functions should ideally have the signature: `(ctx, deps, cfg, logger) (interface{}, error). This makes it easier in the future if we ever need to pass in other information from the bigger component config (name, model, or anything else we add to that config). I wanted this cleanup since I saw an opportunity in this pr.

Reply:
We were also never checking if the converted attributes were okay anywhere, so I added that in the constructor function.